[ramda]: Correct order of reducing-fn params in reduceRight

spellman opened the pull request:
The reducing function in Ramda v0.24.0 reduceRight has signature (elem: T, acc: TResult) => TResult, as per https://github.com/ramda/ramda/blob/v0.24.0/src/reduceRight.js.
Commit where argument order was changed: https://github.com/ramda/ramda/commit/68d1dc19fb035b0fcafcc1857331f9d00986470f

spellman commented:
Well, I'm confused. When I change the order of the parameters in the test functions (inside the two declarations of ```flattenPairs```, lines 330 and 906), ```npm test``` yields ```RangeError: Maximum call stack size exceeded.```

Is v0.24.0 / 0.24.1 of Ramda being used for the tests?

LiamGoodacre commented:
@spellman the "tests" don't actually run, they are only type-checked.  So I guess that the stack error is occurring during typechecking?

mhegazy commented:
looks like you are running into a TS compiler crash. mind filing a bug for this on https://github.com/Microsoft/TypeScript/issues

spellman commented:
No problem. I went over everything again from scratch Friday night -- I thought I was going nuts because this should be straightforward.

spellman commented:
Thanks very much for the help so far!
I started to file the TypeScript bug report but I'm not getting the above error with the latest nightly, typescript@2.6.0-dev.20170826. I suppose that means the problem has been fixed.

Two questions:
1. How does this pull-request proceed? Shall I just wait until a release version of TypeScript compiles the test file without error and then try the tests on Travis?

2. When I run the tests locally with typescript@2.6.0-dev.20170826, I get the same errors on master and with my changes:
```
C:\Users\Cort\Projects\DefinitelyTyped\types\ramda>tsc ramda-tests.ts
ramda-tests.ts(122,19): error TS2345: Argument of type 'Car' is not assignable to parameter of type 'FastCar'.
  Property 'speed' is optional in type 'Car' but required in type 'FastCar'.
ramda-tests.ts(1108,22): error TS2339: Property 'of' does not exist on type 'ArrayConstructor'.
ramda-tests.ts(1109,37): error TS2339: Property 'of' does not exist on type 'ArrayConstructor'.
```
I haven't changed the tsconfig.json; it is
```
{
    "compilerOptions": {
        "module": "commonjs",
        "lib": [
            "es6",
            "dom"
        ],
        "noImplicitAny": true,
        "noImplicitThis": false,
        "strictNullChecks": false,
        "baseUrl": "../",
        "typeRoots": [
            "../"
        ],
        "types": [],
        "noEmit": true,
        "forceConsistentCasingInFileNames": true
    },
    "files": [
        "index.d.ts",
        "ramda-tests.ts"
    ]
}
```
Any ideas as to what I'm doing wrong?

spellman commented:
@mhegazy That bug seems to have been fixed :)

In the test code, I changed the types in the reducing functions [here](https://github.com/spellman/DefinitelyTyped/blob/ramda-reduceRight/types/ramda/ramda-tests.ts#L331) and [here](https://github.com/spellman/DefinitelyTyped/blob/ramda-reduceRight/types/ramda/ramda-tests.ts#L914).
In both cases, ```pairs``` is indeed an array of tuples but the accumulators and return values are the concatenation of those tuples and are therefore of type ```Array<string|number>```.
(Or were those functions named and implemented that way to make a point?)
